nab/detectors/expose/expose_detector.py

```diff
@@ -0,0 +1,71 @@
+import numpy
+
+from sklearn.kernel_approximation import RBFSampler
+
+from nab.detectors.base import AnomalyDetector
+
+
+
+class ExposeDetector(AnomalyDetector):
+
+  """ This detector is an implementation of The EXPoSE (EXPected Similarity
+  Estimation) algorithm as described in Markus Schneider, Wolfgang Ertel,
+  Fabio Ramos, "Expected Similarity Estimation for Lage-Scale Batch and
+  Streaming Anomaly Detection", arXiv 1601.06602 (2016).
+
+  EXPoSE calculates the likelihood of a data point being normal by using
+  the inner product of its feature map with kernel embedding of previous data
+  points. This measures the similarity of a data point to previous points
+  without assuming an underlying data distribution.
+
+  There are three EXPoSE variants: incremental, windowing and decay. This
+  implementation is based on EXPoSE with decay. All three variants have been
+  tried on NAB but decay gives the best results.Parameters for this detector
+  have been tuned to give the best performance.
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(ExposeDetector, self).__init__(*args, **kwargs)
+
+    self.kernel = None
+    self.previousExposeModel = []
+    self.decay = 0.01
+    self.timestep = 1
```

Comment:
Convention is to index from 0, so please start `self.timestep` from 0 and modify the conditional at line 55